daml-assistant/daml-helper/src/DamlHelper.hs

```diff
@@ -126,6 +130,217 @@ withJar jarPath args a = do
 getTemplatesFolder :: IO FilePath
 getTemplatesFolder = fmap (</> "templates") getSdkPath
 
+-- | Initialize a daml project in the current or specified directory.
+-- It will do the following (first that applies):
+--
+-- 1. If the target folder is actually a file, it will error out.
+--
+-- 2. If the target folder does not exist, it will error out and ask
+-- the user if they meant to use daml new instead.
+--
+-- 3. If the target folder is a daml project root, it will do nothing
+-- and let the user know the target is already a daml project.
+--
+-- 4. If the target folder is inside a daml project (transitively) but
+-- is not the project root, it will do nothing and print out a warning.
+--
+-- 5. If the target folder is a da project root, it will create a
+-- daml.yaml config file from the da.yaml config file, and let the
+-- user know that it did that.
+--
+-- 6. If the target folder is inside a da project (transitively) but
+-- is not the project root, it will error out with a message that lets
+-- the user know what the project root is and suggests the user run
+-- daml init on the project root.
+--
+-- 7. If none of the above, it will create a daml.yaml from scratch.
+-- It will attempt to find a Main.daml source file in the project
+-- directory tree, but if it does not it will use daml/Main.daml
+-- as the default.
+--
+runInit :: Maybe FilePath -> IO ()
+runInit targetFolderM = do
+    currentDir <- getCurrentDirectory
+    let targetFolder = fromMaybe currentDir targetFolderM
+        targetFolderRel = makeRelative currentDir targetFolder
+        projectConfigRel
```

Comment:
```suggestion
        projectConfigRel = normalise $ targetFolderRel </> projectConfigName
```
A comment that this is only for display purposes (afaict) would be nice. I spend a few minutes trying to figure out why this is required :smile:

Reply:
Thanks!